src/librustc_mir/borrow_check/nll/type_check/mod.rs

```diff
@@ -993,9 +1005,13 @@ impl<'a, 'gcx, 'tcx> TypeChecker<'a, 'gcx, 'tcx> {
                 // output) types in the signature must be live, since
                 // all the inputs that fed into it were live.
                 for &late_bound_region in map.values() {
-                    self.constraints
-                        .liveness_set
-                        .push((late_bound_region, term_location));
+                    if let Some(borrowck_context) = self.borrowck_context {
+                        let region_vid = borrowck_context.universal_regions.to_region_vid(
+                            late_bound_region);
+                        borrowck_context.constraints
+                            .liveness_constraints
+                            .add_element(region_vid, term_location);
```

Comment:
We wouldn't need to modify all-facts here necessarily -- iirc, polonius only cares about liveness in **variables** -- i.e., regions that exist across many points in the CFG.